detect_secrets/plugins/basic_auth.py

```diff
@@ -5,8 +5,13 @@
 from .base import RegexBasedDetector
 
 
+# This list is derived from RFC 3986 Section 2.2.
```

Comment:
Just posting for posterity/search-ability that an expanded discussion took place in https://github.com/Yelp/detect-secrets/pull/123